### ISSUE TITLE
AppClose event listener never called

### ISSUE THREAD

**Issue by chinenual** [asticode/go-astilectron#267]
I need to call a cleanup function to close streams to ensure some goroutines finish before application stops. I've registered a AppClose listener: 

```go
			w.On(astilectron.EventNameAppClose, func(e astilectron.Event) (deleteListenter bool) {
				log.Printf("Close Event.\n")
				if err = cleanup(); err != nil {
					log.Println(err)
				}
				return false
			})
```
however the listener never gets called 
```
2020/08/09 15:09:13 logger.go:66: App has closed
2020/08/09 15:09:13 logger.go:66: Closing...
2020/08/09 15:09:13 logger.go:66: Stopping...
2020/08/09 15:09:13 logger.go:73: accept tcp 127.0.0.1:63798: use of closed network connection while TCP accepting
#
```
I suspect that since the close events are processed in a goroutine and the stop is called immediately after, the AppClose listeners are never getting a chance to run..? See:

```go
		} else {
			a.l.Debug("App has closed")
			a.dispatcher.dispatch(Event{Name: EventNameAppClose, TargetID: targetIDApp})
		}
		a.dispatcher.dispatch(Event{Name: EventNameAppCmdStop, TargetID: targetIDApp})
```

How can I properly register a listener for application shutdown so I can properly cleanup my app's resources?

**Comment by asticode** [comment]
Unfortunately your assumption is correct, this has been on my TODOs forever but I never got a chance to tackle this issue.

Regarding your situation, is there a reason why you don't do something like this:

```go
defer func() {
  // TODO Close streams or whatever you need to do
}()

bootstrap.Run(bootstrap.Options{...})
```

This way, your `Close` logic is called right after the app closes.

**Comment by chinenual** [comment]
I didn't try it because I was blindly fixated on using an event listener :-).  I've tried added a defer func, but its not (yet?) working.  I'm trying to use fmt to write to stdout to (though I get no error).  Perhaps stdout has already been closed?  More likely I'm doing something in my code to prevent the bootstrap.Run from finishing (I start a goroutine in the OnWait() - I'll see if I can fix this by moving the goroutine initialization elsewhere.)
```go

	defer func() {
		fmt.Printf("Close Event.\n")
		if err = cleanup(); err != nil {
			fmt.Println(err)
		}
	}()```
